pkg/workceptor/workceptor.go

```diff
@@ -631,6 +636,11 @@ func (w *Workceptor) scanForUnits() {
 				}
 				w.activeUnits[fi.Name()] = unit
 				if si.Node != "" && si.Node != w.nc.NodeID() {
+					if si.LocalCancel {
```

Comment:
on control node restart, this will make sure we restart the cancellation goroutine